test/Confluent.Kafka.IntegrationTests/Tests/Consumer_Poll.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+
+namespace Confluent.Kafka.IntegrationTests
+{
+    public static partial class Tests
+    {
+        /// <summary>
+        ///     Basic Consumer test (poll mode).
+        /// </summary>
+        [Theory, MemberData(nameof(KafkaParameters))]
+        public static void Consumer_Poll(string bootstrapServers, string topic)
+        {
+            int N = 2;
+            var firstProduced = Util.ProduceMessages(bootstrapServers, topic, 100, N);
+
+            var consumerConfig = new Dictionary<string, object>
+            {
+                { "group.id", "simple-produce-consume" },
+                { "bootstrap.servers", bootstrapServers },
+                { "session.timeout.ms", 6000 }
+            };
+
+            using (var consumer = new Consumer(consumerConfig))
+            {
+                int msgCnt = 0;
+                bool done = false;
+
+                consumer.OnMessage += (_, msg) =>
+                {
+                    Assert.Equal(msg.Error.Code, ErrorCode.NO_ERROR);
+                    msgCnt += 1;
+                };
+
+                consumer.OnPartitionEOF += (_, partition)
+                    => done = true;
+
+                consumer.OnPartitionsAssigned += (_, partitions) =>
+                {
+                    Assert.Equal(partitions.Count, 1);
```

Comment:
What is setting up the topic to only have one partition?

It might be better not to have such strict requirements on the test cluster but instead do something like: is better 
`if (++eofCnt == currentAssignment.Length) done = true;`